### PR TITLE
samples: drivers: counter: alarm: Fix compilation for nrf boards

### PR DESCRIPTION
**Pull request by nordic-krch**
Sample was not compiling for nordic boards because RTC_0
instance was not enabled by default and RTC_0 device is using
different DT define.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

File: samples/drivers/counter/alarm/src/main.c [hunk 1]
```diff
@@ -13,6 +13,15 @@
 #define DELAY 2000000
 #define ALARM_CHANNEL_ID 0
 
+#if defined(DT_RTC_0_NAME)
+#define COUNTER_NAME DT_RTC_0_NAME
+#elif defined(DT_NORDIC_NRF_RTC_RTC_0_LABEL)
+#define COUNTER_NAME DT_NORDIC_NRF_RTC_RTC_0_LABEL
```
Comment (mbolivar) [inline samples/drivers/counter/alarm/src/main.c]: Can't we do this with an alias in our soc DTSI?

File: samples/drivers/counter/alarm/prj.conf [hunk 2]
```diff
@@ -1,2 +1,4 @@
 CONFIG_PRINTK=y
 CONFIG_COUNTER=y
+CONFIG_COUNTER_RTC0=y
```
Comment (mbolivar) [inline samples/drivers/counter/alarm/prj.conf]: Could this be done conditionally in the application Kconfig?

```
config COUNTER_RTC0
	default y if SOC_FAMILY_NRF
```

Something like that.

File: samples/drivers/counter/alarm/src/main.c (hunk 1 above)
Comment (nordic-krch) [reply samples/drivers/counter/alarm/src/main.c]: added to `dts_fixup.h`

File: samples/drivers/counter/alarm/prj.conf (hunk 2 above)
Comment (nordic-krch) [reply samples/drivers/counter/alarm/prj.conf]: done

File: samples/drivers/counter/alarm/src/main.c (hunk 1 above)
Comment (anangl) [reply samples/drivers/counter/alarm/src/main.c]: `DT_NORDIC_NRF_RTC_RTC_0_LABEL` is actually generated from an alias, such as [this one](https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/nordic/nrf52840.dtsi#L42). You can't (currently) get `DT_RTC_0_NAME` generated by using an alias defined in DTS. The only way is to add such fixups for our SoCs (and although a bit ugly, this seems to be the best way to solve this at the moment).

**Review by anangl — APPROVED** [review]

**Review by mbolivar — APPROVED** [review]
Thanks for the explanations about aliases and fixups. LGTM.

File: soc/arm/nordic_nrf/nrf91/dts_fixup.h [hunk 3]
```diff
@@ -32,10 +32,13 @@
 #define DT_SPI_2_NAME			DT_NORDIC_NRF_SPI_SPI_2_LABEL
 #define DT_SPI_3_NAME			DT_NORDIC_NRF_SPI_SPI_3_LABEL
 
-#define DT_WDT_0_NAME                   DT_NORDIC_NRF_WATCHDOG_WDT_0_LABEL
+#define DT_WDT_0_NAME			DT_NORDIC_NRF_WATCHDOG_WDT_0_LABEL
 
-#define DT_TIMER_0_NAME                 DT_NORDIC_NRF_TIMER_TIMER_0_LABEL
-#define DT_TIMER_1_NAME                 DT_NORDIC_NRF_TIMER_TIMER_1_LABEL
-#define DT_TIMER_2_NAME                 DT_NORDIC_NRF_TIMER_TIMER_2_LABEL
+#define DT_TIMER_0_NAME			DT_NORDIC_NRF_TIMER_TIMER_0_LABEL
```
Comment (ioannisg) [inline soc/arm/nordic_nrf/nrf91/dts_fixup.h]: Looks like indentation is not correct here.

Comment (nordic-krch) [reply soc/arm/nordic_nrf/nrf91/dts_fixup.h]: nope, i've corrected it (previously those entries had spaces instead of tabs like the others).

Comment (ioannisg) [reply soc/arm/nordic_nrf/nrf91/dts_fixup.h]: Yeap, weird github issue, i think.

**Review by ioannisg — APPROVED** [review]

**Comment by nordic-krch** [comment]
@carlescufi @nashif can we merge it?

**Pull request merged**